doc/interfaces/multicast_ip.rst

```diff
@@ -0,0 +1,57 @@
+.. _multicast_ip_doc:
+
+Multicast IP Interface
+======================
+
+This module implements transport of CAN and CAN FD messages over UDP on Multicast IPv4 and IPv6.
```

Comment:
I agree that multicast is very handy, but I have one question about this:

- How is CAN arbitration handled? Is there any specific order in which CAN messages are received by all participating CAN "controllers" on the multicast IP bus? For example, in a real life CAN bus, CAN messages have an arbitration function, so CAN messages with a lower ID have higher priority. Also, on real hardware CAN messages are received in the same order at all CAN controllers. This might seem details, so I'm not sure how important this is for simulation purpose. But it might trigger some unexpected situations when higher level protocols make some assumptions about this.

As a side note: I created [virtualcan](https://github.com/windelbouwman/virtualcan), in which CAN bus arbitration is not yet implemented, but since this project does take the central server approach, it is possible to add priority sorting, and messages will arrive at all ends in the same order.

It might be good to document how the priority / ordering of CAN message is handled so that people know what to expect.

Reply:
Sure, at the very least this should be documented. I actually didn't think about that until right now. Huh.

But the implementation simply receives whatever is coming in that order, and does not care about overlapping timestamps or something like that. It does not entirely faithfully reproduce a physical network, that's correct. But if the bus isn't near capacity, that shouldn't be a problem, right?

This interface would have to implement throttling too in order to be close to realistic. Right now, we could theoretically send Gigabits of traffic over that "CAN" bus.

I'd say it's okay if the virtual interfaces are idealized in that way. The `python-can` *virtual* interface does not care either (and we should state that in the docs too).

Reply:
> But if the bus isn't near capacity, that shouldn't be a problem, right?

Let's hope so :wink: . This should be documented at the very least indeed. The thing with these kind of assumptions is that in 99.99% of the cases this poses no issue, and once a month, due to a glitch there is some weird failure. This might be troublesome to pinpoint.. I'd say, put the limitations in the docs, and maybe even issue a warning during opening of the interface?

> This interface would have to implement throttling too in order to be close to realistic. Right now, we could theoretically send Gigabits of traffic over that "CAN" bus.

Throttling would be a nice feature to add indeed! I'll keep it in mind.

> The `python-can` _virtual_ interface

The term virtualcan is a bit overloaded. Thanks for naming this plugin `virtualcan_multicast`!

Reply:
> The term virtualcan is a bit overloaded. 

Yeah. ;) It's actually `multicast_ip` to specify the carrying layer. Would you say `udp_multicast` would be better?